docs/e2e-test.md

```diff
@@ -26,3 +26,30 @@ After a finished test, cleanup is performed automatically. If the test setup fai
 ```sh
 make clean -C /opt/metal3-dev-env/metal3-dev-env/ && sudo rm -rf /opt/metal3-dev-env/
 ```
+
+## Included tests
+
+The e2e tests currently run one after the other, using the same environment for all of them.
+They run in this order:
+
+1. Remediation
+1. Inspection¹
+1. Pivoting
+1. Upgrade BMO
+1. Upgrade Ironic
+1. Certificate rotation
+1. Node reuse
+1. Re-pivoting
+
+¹ Inspection is actually run [in the middle of the remediation test](https://github.com/metal3-io/cluster-api-provider-metal3/blob/8d08f375de93a793f839b42b5ec40e6bebf98664/test/e2e/remediation_test.go#L108) for practical reasons at the moment.
+
+The ephemeral cluster is first launched using [metal3-dev-env](https://github.com/metal3-io/metal3-dev-env).
+The remediation and inspection tests are then run with the controllers still in the ephemeral cluster.
+After this, the controllers are moved to the target cluster during pivoting.
+From this point on all tests are run in the target cluster, until the controllers are moved back to the ephemeral cluster during re-pivoting.
+
+### Upgrade tests
+
+Both Ironic and BMO upgrade tests currently only check that a new version can be rolled out (by going from `latest` to `main`).
+However, they do not actually upgrade from some older release to a newer since they are not yet integrated with the full upgrade test in CAPM3.
+Once this is in place, the idea is to upgrade Ironic and BMO together with CAPM3 from the previous minor release to the latest.
```

Comment:
As discussed offline, I would suggest removing **### Upgrade tests** sections since it might confuse others thinking that those tests are run when e2e upgrade tests are triggered since you are adding e2e upgrade trigger phrases under `Upgrade tests` in the root readme file. 
Instead, we could:

- add Ironic and BMO upgrade related just after feature tests description on line 50 for now since they are part of the  `normal e2e tests`
- once we get Ironic and BMO tests out of `normal e2e tests`  and include them in the `e2e upgrade tests`, we can put those notes + management cluster upgrade related notes back again here.

Reply:
Updated, please take a look!

Reply:
Looks good, thanks!